#!/usr/bin/env python
# coding=UTF-8

PACKAGE = "set_mode"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#添加参数 （name , typt, level, description, default, min,max）
gen.add("catch_point",   bool_t,   0, "catch point",  True)
gen.add("add_path",   bool_t,   0, "add_path_mode",  False)
gen.add("delete_point",   bool_t,   0, "delete_point_mode",  False)
gen.add("delete_path",   bool_t,   0, "delete_path_mode",  False)

#size_enum = gen.enum([ gen.const("Low",      int_t, 0, "Low is 0"),
#                       gen.const("Medium",   int_t, 1, "Medium is 1"),
#                       gen.const("High",     int_t, 2, "Hight is 2")],
#                     "Select from the list")

#gen.add("size", int_t, 0, "Select from the list", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "set_mode", "mode"))
